### PR TITLE
Add Subscription#push_config

### LINKED ISSUE #3048: PubSub: Support authentication_method / oidc_token
Please add support for Pub/Sub [`Subscription.PushConfig.authentication_method`](https://github.com/googleapis/googleapis/blob/eae7fe03e6e0ca90d0a3e20211f31dca6fb2cb87/google/pubsub/v1/pubsub.proto#L672)

### PR DESCRIPTION
**Pull request by blowmage**
Add `Subscription#push_config` and `Subscription::PushConfig`. This exposes more information about how the subscription pushes to an external HTTP endpoint, including setting authentication on the call to the endpoint (only OidcToken for now), and what API version to use for formatting the data sent.

[closes #3048]

### CODE REVIEW

**Comment by blowmage** [comment]
@googleapis/yoshi-ruby This PR has been updated and is passing CI now. PTAL.

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb [hunk 1]
```diff
@@ -218,6 +219,44 @@ def endpoint= new_endpoint
...
+        # using the method's block. Unlike {#endpoint}, which will retrieve the
+        # full resource from the API before returning. To get the actual values
+        # for a reference object, call {#reload!} before calling {#push_config}.
+        #
+        # @yield [push_config] a block for modifying the push configuration
+        # @yieldparam [Subscription::PushConfig] push_config the push
+        #   configuration
+        #
+        # @return [Subscription::PushConfig]
+        #
```
Comment (quartzmo) [inline google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb]: Code examples?

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription/push_config.rb [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    module PubSub
+      class Subscription
+        # Configuration for a push delivery endpoint.
+        class PushConfig
+          # @private
+          def initialize
+            @grpc = Google::Cloud::PubSub::V1::PushConfig.new
+          end
+
+          # A URL locating the endpoint to which messages should be pushed.
```
Comment (quartzmo) [inline google-cloud-pubsub/lib/google/cloud/pubsub/subscription/push_config.rb]: I would add: "For example, a Webhook endpoint might use "https://example.com/push"." from [here](https://cloud.google.com/pubsub/docs/reference/rpc/google.pubsub.v1#pushconfig).

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription/push_config.rb [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+# limitations under the License.
+
+
+require "google/pubsub/v1/pubsub_pb"
+
+module Google
+  module Cloud
+    module PubSub
+      class Subscription
+        # Configuration for a push delivery endpoint.
```
Comment (quartzmo) [inline google-cloud-pubsub/lib/google/cloud/pubsub/subscription/push_config.rb]: Shouldn't we stick with the two-hash header we usually use for docs comments?

```
##
# Configuration...
```

Comment (quartzmo) [reply google-cloud-pubsub/lib/google/cloud/pubsub/subscription/push_config.rb]: Code examples?

Comment (blowmage) [reply google-cloud-pubsub/lib/google/cloud/pubsub/subscription/push_config.rb]: Added comment headers.

Comment (blowmage) [reply google-cloud-pubsub/lib/google/cloud/pubsub/subscription/push_config.rb]: Added code examples.

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription/push_config.rb (hunk 2 above)
Comment (blowmage) [reply google-cloud-pubsub/lib/google/cloud/pubsub/subscription/push_config.rb]: Added.

File: google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb (hunk 1 above)
Comment (blowmage) [reply google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb]: Added.

**Pull request merged**